WooCommerce/Classes/Tools/UnitInputFormatter/ShippingInputFormatter.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// `UnitInputFormatter` implementation for shipping input.
+///
+struct ShippingInputFormatter: UnitInputFormatter {
+
+    // Match all the characters not included in the set: numbers between 0 and 9, and - . , %
+    let regex = "[^-0-9%.,]"
```

Comment:
also this might be a change in WC 3.9, `,` isn't allowed in wp-admin anymore:

<img width="318" alt="Screen Shot 2020-01-28 at 11 00 27 AM" src="https://user-images.githubusercontent.com/1945542/73232448-3eb71900-41be-11ea-8e9e-cc2ac8dabed2.png">

Reply:
I used the validation used on the frontend side but I think that the backend removes the `%`, so I'm going to remove it from the regex, but I'm not sure if we can remove the comma `,` since in the original regex was used the decimal separator configured in the website settings, and since we support also previous versions of WC it's a little bit tricky. I also see that WC 3.9 uses [this regex](https://github.com/woocommerce/woocommerce/pull/24281#pullrequestreview-328111263) `/\.(?![^.]+$)|[^0-9.]/` to validate the field in the backend, so a value like `9.99.00` now is not admitted, but before it was.

Reply:
> so a value like 9.99.00 now is not admitted, but before it was

@pmusolino Are we going to implement this in here then? Or have you already done it?

Reply:
Sorry, that was an irresponsible question. I tested it and it looks like we still support `"9.99.0"`.

Reply:
Now a value like `9.99.0` is valid but formatted it will be `999.0` because while typing now we accept only one decimal separator.
I added also some new tests for these cases.